src/NServiceBus.Persistence.Sql.Analyzer/SagaMetadataGenerator.cs

```diff
@@ -0,0 +1,243 @@
+﻿namespace NServiceBus.Persistence.Sql.Analyzer;
+
+using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Operations;
+
+[Generator]
```

Comment:
Mark it as generating csharp?

Reply:
> Mark it as generating csharp?

That's the default (below is from GeneratorAttribute source) … is it necessary?

```
public GeneratorAttribute()
    : this(LanguageNames.CSharp) { }
```

> split it into emitter and parser similar to the core ones to have a consistent shape

There the benefit is the overlap between AddHandler and AddSaga. What's the benefit here?